include/rocksdb/manifest_ops.h

```diff
@@ -0,0 +1,30 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#include <cassert>
+
+#include "rocksdb/env.h"
+#include "rocksdb/file_checksum.h"
+#include "rocksdb/status.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+Status GetCurrentManifestPath(const std::string& abs_path, FileSystem* fs,
```

Comment:
Public APIs get extra scrutiny for usefulness, clarity, sustainability, etc.

I'm not really opposed to this function going into the public API, but the default answer to "does this function belong in the public API?" is no, because it entails a degree of ongoing support and complicates future refactorings.

By the way, the `abs_path` parameter is, for better or worse, known as `name` or `db_name` in our public API even though it is a directory.

It's not clear what `is_retry` is for exactly.

Pursuant to an old Google coding style guideline, we tend to use `NonPolymorphicType*` for output parameters, and they are often self-explanatory.

Although there are lots of places where we idiomatically use raw pointers for polymorphic input or input/output parameters that are non-nullable and do not "escape" the function call, I consider this an anti-pattern. `FileSystem& fs` is more clear about the intent and calling conventions.

Reply:
Thanks for the helpful context, Peter!

1. Moved the `manifest_ops.h` from public facing API to internal API under the `db/` directory, mimicking the `experimental` and `convenience` files.
2. Renamed `abs_path` to `dbname` as per broad codebase convention.
3. `is_retry` parameter is used to toggle the `verify_and_reconstruct_read` flag which indicates the occurrence of the corruption in a previous read and expresses the callers intent to re-read the file with stronger guarantees. I went through the codebase and it appears that the only place where this mode is _potentially_ used is in `DBImpl::Recover` as a best effort to obtain the MANIFEST file. I'll add more comment on the usage. 
4. Wrt `NonPolymorphicType*` comment about the output parameters, those relate to the `manifest_path` (`string*`) and `manifest_file_number` (`uint64_t*`). Are there any specific types you'd want to see instead?
5. Wrt `FileSystem& fs`, the explanation makes total sense. However, given the 'tech-debt' in a form of downstream callers (ex. `ReadFileToString`), I'd prefer to keep it as is. Let me know if you have strong preference about this one particular case, Peter. Happy to refactor if needed.